File: Year-3/Semester-2/Technology-of-Software-Production-Course-Project/tuvarna-ecommerce-system/Models/DTOs/UserCreateDTO.cs
using System.ComponentModel.DataAnnotations;

namespace tuvarna_ecommerce_system.Models.DTOs
{
    public class UserCreateDTO
    {

        [Required(ErrorMessage = "Username is required.")]
        [StringLength(50, ErrorMessage = "Username must not exceed 50 characters.")]
        public string Username { get; set; }

        [Required(ErrorMessage = "Email is required.")]
        [EmailAddress(ErrorMessage = "A valid email address is required.")]
        [StringLength(100, ErrorMessage = "Email must not exceed 100 characters.")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Password is required.")]
        [StringLength(100, MinimumLength = 6, ErrorMessage = "Password must be between 6 and 100 characters long.")]
        public string Password { get; set; }

        [Required(ErrorMessage = "Role is required.")]
        public string Role { get; set; }
    }
}

File: Year-3/Semester-2/Internet-Technologies/Lab-06/src/main/java/bg/tuvarna/sit/dto/UserCreateDto.java
package bg.tuvarna.sit.dto;

import bg.tuvarna.sit.entities.RoleType;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.util.HashSet;
import java.util.Set;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UserCreateDto {

  @NotBlank(message = "Name is required and cannot be blank")
  private String name;

  @NotBlank(message = "Email is required and cannot be blank")
  @Email(message = "Email should be valid")
  private String email;

  @NotBlank(message = "Username is required and cannot be blank")
  private String username;

  @NotBlank(message = "Password is required and cannot be blank")
  @Size(min = 8, message = "Password must be at least 8 characters long")
  private String password;

  @NotNull(message = "Roles set cannot be null")
  @Size(min = 1, message = "At least one role is required")
  private Set<RoleType> roles = new HashSet<>();
}
